processor/metrics.go

```diff
@@ -61,17 +62,39 @@ var (
 		stats.UnitDimensionless)
 )
 
+// SpanCountStats represents span count stats grouped by service if DETAILED telemetry level is set,
+// otherwise only overall span count is stored in serviceSpansCounts.
+type SpanCountStats struct {
+	processorName      string
+	serviceSpansCounts map[string]int
+	allSpansCount      int
+	isDetailed         bool
+}
+
+func NewSpanCountStats(td pdata.Traces, processorName string) *SpanCountStats {
+	scm := &SpanCountStats{
+		processorName: processorName,
+		allSpansCount: td.SpanCount(),
+	}
+	if serviceTagsEnabled() {
+		scm.serviceSpansCounts = td.SpanCountByResourceStringAttribute(conventions.AttributeServiceName)
```

Comment:
Do we really need this? I suspect this is going to have significant performance impact. I would like to see a perf showing how much time and memory this takes and assess if this is needed at all.
Why do we need to collect per service stats? Is that a collector job? Is this a queue's job?
Looks like we are doing the job that the backend should be doing. If we really need it let's make it a separate processor.
I may be missing something, please clarify.

Reply:
Just to be clear. I understand this was an existing functionality. But I do not understand why did we have this in the queued_retry processor.

Reply:
Probably because the queued_retry processor is the step where we expect to drop spans. So we want to show how many of them were received and dropped. But I agree that showing which service was affected by drops is not that super-important. And I'd rather simplify the implementation and do not report service based metrics. 

@pjanotti do you know how actively these service grouped metrics being used/

cc @bogdandrutu

Reply:
Yes, some have used it in the past to identify which services were generating expected high volume of data. That said this is not available on the default metric configuration one has to select "detailed" to see them.

Reply:
As discussed offline we are OK to keep this functionality since it is not enabled by default.